### PR TITLE
Document a little about moderators and tiers

### PR DESCRIPTION
**Pull request by jonmeow**
Note, partly writing this down to make sure everyone's in agreement with what's written.

### CODE REVIEW

**Review by CelineausBerlin — APPROVED** [review]

File: docs/project/moderators.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+## Becoming a moderator
+
+TODO: Need to figure out right process for asking. (that is, how?)
+
+The Conduct team reviews new moderators. Community members will be considered
+mainly based on their history on the Carbon Language Discord, although
+applicants may also note history on other servers.
+
+Prior to being granted moderation privileges, new moderators will be join a
```
Comment (geoffromer) [inline docs/project/moderators.md]:
```suggestion
Prior to being granted moderation privileges, new moderators will need to attend a
```

File: docs/project/moderators.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+## Moderation powers
+
+There is a moderation playbook for moderators with instructions on how to use
+these powers.
+
+-   **Discord**
+
+    -   **Delete message** (senior): Delete a message owned by another user.
+
```
Comment (chandlerc) [inline docs/project/moderators.md]: Just me, but I'd generally expect all of the "delete message" and "edit message" capabilities to be roughly the same tier. Below, they are "all", which I think make sense.

The "senior" capabilities would then be specifically ones with a larger scope of impact -- entire conversation, channel, or persistent user effect.

File: docs/project/moderators.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    -   **Edit title** (all): Edits the title of a thread.
+
+    -   **Delete message** (all): Removes a single message from a conversation.
+        Hiding or editing messages is preferred.
+
+        -   This cannot be done to the first message in a conversation.
+
+    -   **Delete conversation** (all): Deletes a full GitHub conversation.
+        Hiding or editing messages is preferred.
```
Comment (chandlerc) [inline docs/project/moderators.md]: Maybe this should be "senior"?

**Review by chandlerc — COMMENTED** [review]
Good to document all this! Added some thoughts about the nuance below, but happy if other moderators prefer a different split here. Don't block on me. =]

File: docs/project/moderators.md (hunk 2 above)
Comment (jonmeow) [reply docs/project/moderators.md]: You're not specific so I assume you're not familiar with GitHub's ACL controls. Unlike Discord, GitHub is more granular with permissions. As far as I know, the permissions like "Lock conversation" or "Hide message" cannot be granted separately from "Delete conversation" or "Delete message".

So here are what I see as the options:

1. Only allow _senior_ moderators permissions on GitHub. Regular moderators will not have any special permissions.
2. Grant all moderators permission to delete messages on Discord.
3. Leave this as is.

I think what you're suggesting is closest to (1), requiring regular moderators to ask for a senior moderator to help with issues on GitHub. Is that your intent?

File: docs/project/moderators.md (hunk 3 above)
Comment (jonmeow) [reply docs/project/moderators.md]: See above

File: docs/project/moderators.md (hunk 2 above)
Comment (chandlerc) [reply docs/project/moderators.md]: No, the granularity wasn't at all clear to me and that led to my suggestion. If there is no ability to have different controls between locking and hiding (non-destructive) and editing and deleting (destructive), not sure I have strong opinions about which option we use. I'll leave it to you and other moderators.


Regardless, maybe change the structure of the GitHub section to make it clear that there's really just a single control here that covers all of these operations? I had assumed the bullet points here roughly reflected the granularity that these roles can be set up.

Comment (jonmeow) [reply docs/project/moderators.md]: > locking and hiding (non-destructive) and editing and deleting (destructive),

On some further thought, hiding looks separate, so I've reduced to that at least. I don't think locking is an option.

> Regardless, maybe change the structure of the GitHub section to make it clear that there's really just a single control here that covers all of these operations?

Added a note beneath the permissions; I hadn't intended this page to be documenting the implementation, mainly the consequences. i.e., more a reference for moderators than for admin.

File: docs/project/moderators.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+_senior moderators_, and will have extra privileges to help address harmful
+conduct. Senior moderators will have the "senior-moderator" role on Discord.
+
+## Becoming a moderator
+
+TODO: Need to figure out right process for asking. (that is, how?)
+
+The Conduct team reviews new moderators. Community members will be considered
+mainly based on their history on the Carbon Language Discord, although
+applicants may also note history on other servers.
```
Comment (chandlerc) [inline docs/project/moderators.md]: Minor tweak to include non-Discord stuff?

```suggestion
mainly based on their history in Carbon Language community spaces (Discord, GitHub, etc.), although
applicants may also note history in other communities.
```

**Review by chandlerc — APPROVED** [review]
Still a little hesitant about the terminology here, but figure we can also fix forward if needed.

**Pull request merged**